### ISSUE TITLE
misleading error message when updating mismatched product with mismatched attributes-posm entries

### ISSUE THREAD

**Issue by torvista** [lat9/options_stock_support#43]
This may be a right-over-the edge case, but anyway...

Culprit is a spare/retired product OCG0038.
When updating the options.stock via DBIO, it was complaining about this one.
I assumed it was due to the model numbers being duplicated with OCG0037.
So, when attempting to change the model numbers from OCG0037->38, I got this error (once corrected as per #8 ).

![posm error](https://user-images.githubusercontent.com/4391026/97925760-3cdc4080-1d62-11eb-8d11-9e0fb88eddd8.gif)

Note that the image shows the posm models as OCG0038...i.e. having been corrected, but in truth they still remain at OCG0037.

Debug is

> Request URI: /ADMIN/products_options_stock.php?action=update
> --> PHP Notice: Undefined offset: 3230 in /ADMIN/products_options_stock.php on line 1168.

which is 
`$pos_quantity = $_POST['pos_quantity'][$pos_id];`

But actually the posm options were mismatched to its attributes.
![attribute](https://user-images.githubusercontent.com/4391026/97924625-3f3d9b00-1d60-11eb-9556-8236be023ff2.gif)

with an orphan entry.

So, a misleading error error message.

**Comment by lat9** [comment]
Agreed on the bug, but (as noted by your comment) it's a very edge-case ... thus the 'low priority' label.

**Comment by lat9** [comment]
That'll be corrected in POSM v4.2.2.  In the /admin/products_options_stock.php, find this code block:
```php
        if ($info_array['valid']) {
            $additional_class = ($info_array['quantity'] <= $posm_stock_reorder_level || $oos_date_is_expired) ? ' lowstock' : '';
            $quantity_parms = 'class="quantity"';
        } else {
            $additional_class = ' removed';
            $quantity_parms = ' class="quantity" disabled';
        }
```

and change to
```php
        if ($info_array['valid']) {
            $additional_class = ($info_array['quantity'] <= $posm_stock_reorder_level || $oos_date_is_expired) ? ' lowstock' : '';
            $quantity_parms = 'class="quantity"';
        } else {
            $additional_class = ' removed';
            $quantity_parms = ' class="quantity" readonly';
        }
```
